woodwork/column_accessor.py

```diff
@@ -246,6 +248,18 @@ def wrapper(*args, **kwargs):
                                                                                           invalid_schema_message,
                                                                                           'Series')
                         warnings.warn(warning_message, TypingInfoMismatchWarning)
+                elif _is_dataframe(result):
```

Comment:
Are there other examples of Series calls that return a DataFrame? Both ones that will and will not have a valid woodwork schema

Because we're adding functionality for those DataFrames returned that can have a valid schema, and I don't think we'll be losing functionality for non woodwork dataframes removed since we still return the DataFrame, but it'd be nice to confirm that.

Reply:
I'm not sure how many exist, but calling `series.reset_index(drop=False)` also returns a frame, which will have an invalid schema. I added this case to the test and renamed the test to reflect the increased scope.